src/ntcore_c.cpp

```diff
@@ -436,3 +436,257 @@ void NT_DisposeRpcCallInfo(NT_RpcCallInfo *call_info) {
   NT_DisposeString(&call_info->name);
   NT_DisposeString(&call_info->params);
 }
+
+/* Interop Utility Functions */
+
+/* Array and Struct Allocations */
+
+/* Allocates an integer or boolean array of the specified size. */
+int *NT_AllocateBooleanArr(size_t size)
+{
+	int *retVal = static_cast<int*>(
+		std::malloc(size * sizeof(int)));
+	return retVal;
+}
+
+/* Allocates a double array of the specified size. */
+double *NT_AllocateDoubleArr(size_t size)
+{
+	double *retVal = static_cast<double*>(
+		std::malloc(size * sizeof(double)));
+	return retVal;
+}
+
+/* Allocates an NT_String array of the specified size. */
+struct NT_String *NT_AllocateNTStringArr(size_t size)
+{
+	NT_String *retVal = static_cast<NT_String*>(
+		std::malloc(size * sizeof(NT_String)));
+	return retVal;
+}
+
+/* Allocates */
+struct NT_String NT_AllocateNTString(size_t size)
+{
+	NT_String retVal;
+	retVal.len = size;
+	retVal.str = static_cast<char*>(
+		std::malloc(size * sizeof(char)));
+	return retVal;
+}
+
+void NT_FreeDoubleArr(double *v_double)
+{
+	std::free(v_double);
+}
+void NT_FreeBooleanArr(int *v_boolean)
+{
+	std::free(v_boolean);
+}
+void NT_FreeStringArr(struct NT_String *v_string, size_t arr_size)
+{
+	for (size_t i = 0; i < arr_size; i++)
```

Comment:
NT_DisposeValue should free any internals because the code allocates them in NT_GetEntryValue, the most common use case.  In general I like freeing internals, although it does risk double-frees if the user frees it themselves without setting the pointer to null afterwards.

One thing that should be cleaned up in the C API is that only DisposeConnectionInfoArray frees the passed pointer while all other functions do not, which is confusing (while the code allocates the ConnectionInfo array and thus should be responsible for freeing it, the function probably shouldn't use the Dispose* name).